raiden/api/python.py

```diff
@@ -137,47 +141,62 @@ def channel_open(
             settle_timeout=None,
             reveal_timeout=None,
             poll_timeout=DEFAULT_POLL_TIMEOUT):
-        """ Open a channel with the peer at `partner_address`
-        with the given `token_address`.
-        """
-        if reveal_timeout is None:
-            reveal_timeout = self.raiden.config['reveal_timeout']
 
-        if settle_timeout is None:
-            settle_timeout = self.raiden.config['settle_timeout']
+        # Prevent concurrent attempts to open a channel with the same token and
+        # partner address
+        if partner_address not in self.open_channel_locks:
+            new_open_lock = gevent.lock.Semaphore()
+            self.open_channel_locks[(token_address, partner_address)] = new_open_lock
```

Comment:
`RaidenAPI` is too high in the stack for it to control synchronization, these locks should go in the [`ChannelManager` proxy](https://github.com/raiden-network/raiden/blob/1f17a15c518a4af65f03c015cf13c63ef755150d/raiden/network/proxies/channel_manager.py#L46). The reason is that internal code /may/ use the proxies directly, thus the synchronization would be skipped.